src/generator/protocol/operations.ts

```diff
@@ -100,22 +100,34 @@ function createProtocolRequest(client: string, op: Operation, imports: ImportMan
   text += `func (${client}) ${name}(${generateParamsSig(sig.protocolSigs.requestMethod.params)}) (${sig.protocolSigs.requestMethod.returns.join(', ')}) {\n`;
   text += `\tu.Path = path.Join(u.Path, "${op.request.protocol.http!.path}")\n`;
   const reqObj = `azcore.NewRequest(http.Method${pascalCase(op.request.protocol.http!.method)}, u)`;
-  if (getMediaType(op.request.protocol) === 'none') {
+  const headerParamCount = values(op.request.parameters).where((each: Parameter) => { return each.protocol.http!.in === 'header'; }).count();
+  if (getMediaType(op.request.protocol) === 'none' &&  headerParamCount == 0) {
     // no request body so nothing to marshal
     text += `\treturn ${reqObj}, nil\n`;
   } else {
     const bodyParam = values(op.request.parameters).where((each: Parameter) => { return each.protocol.http!.in === 'body'; }).first();
     text += `\treq := ${reqObj}\n`;
     // default to the body param name
-    let body = bodyParam!.language.go!.name;
-    if (bodyParam!.schema.type === SchemaType.Constant) {
-      // if the value is constant, embed it directly
-      body = formatConstantValue(<ConstantSchema>bodyParam!.schema);
+    // have to check if bodyParam is null since we can enter the else if there are headers that need to be added to the request
+    if (bodyParam != null) {
+      let body = bodyParam!.language.go!.name;
+      if (bodyParam!.schema.type === SchemaType.Constant) {
+        // if the value is constant, embed it directly
+        body = formatConstantValue(<ConstantSchema>bodyParam!.schema);
+      }
+      text += `\terr := req.MarshalAs${getMediaType(op.request.protocol)}(${body})\n`;
+      text += `\tif err != nil {\n`;
+      text += `\t\treturn nil, err\n`;
+      text += `\t}\n`;
+    }
+    // add specific request headers and import the fmt package only if there are headers to add
+    if (headerParamCount > 0) {
+      imports.add("fmt");
+      const headerParam = values(op.request.parameters).where((each: Parameter) => { return each.protocol.http!.in === 'header'; });
+      headerParam.forEach(header => {
+        text += `\treq.Header.Set("${header.language.go!.name}", fmt.Sprintf("%v", ${header.language.go!.name}))\n`;
```

Comment:
I think you can use `formatQueryParamValue()` (and rename it) instead of the `Sprintf` here since we already know the parameter type (plus it will be necessary in cases where we need to format the time in a specific way).

Reply:
Ok will do, but does formatQueryParamValue already exist? I couldn't find it doing a quick search right now, aside from formatConstantValue

Reply:
Yeah it was merged in as part of my query params work.

Reply:
Ah cool, will do 👍